compiler/src/iree/compiler/API2/Embed.h

```diff
@@ -31,7 +28,7 @@
 #endif
 #else
 // Non-windows: use visibility attributes.
-#define IREE_EMBED_EXPORTED __attribute__((visibility("default")))
+#define IREE_EMBED_EXPORTED [[gnu::visibility("default")]]
```

Comment:
There are build errors coming from this:

clang: https://github.com/iree-org/iree/actions/runs/3850751263/jobs/6561256659
```
In file included from /work/compiler/src/iree/compiler/API2/Stub/LoaderTest.c:12:
/work/compiler/src/iree/compiler/API2/Embed.h:56:1: error: expected expression
IREE_EMBED_EXPORTED void ireeCompilerErrorDestroy(iree_compiler_error_t *error);
^
/work/compiler/src/iree/compiler/API2/Embed.h:31:30: note: expanded from macro 'IREE_EMBED_EXPORTED'
#define IREE_EMBED_EXPORTED [[gnu::visibility("default")]]
                             ^
```

gcc: https://github.com/iree-org/iree/actions/runs/3850751263/jobs/6561257330
```
In file included from /work/compiler/src/iree/compiler/API2/Stub/LoaderTest.c:12:
/work/compiler/src/iree/compiler/API2/Embed.h:56:1: error: expected identifier or '(' before '[' token
   56 | IREE_EMBED_EXPORTED void ireeCompilerErrorDestroy(iree_compiler_error_t *error);
      | ^~~~~~~~~~~~~~~~~~~
```